src/main/webapp/app/text-assessment/text-assessment.component.html

```diff
@@ -124,12 +125,19 @@ <h4 class="col-12" jhiTranslate="artemisApp.textAssessment.detail.feedback">Feed
         <div *ngIf="!referencedFeedback || referencedFeedback.length == 0" class="col-12 col-lg-8 col-xl-6">
             <div class="alert alert-secondary text-center" role="alert">
                 <p jhiTranslate="artemisApp.textAssessment.assessInstruction">Please highlight the text block you want to assess and click the "Assess" button.</p>
-                <a class="alert-link" jhiTranslate="artemisApp.textAssessment.predefineTextBlocks" (click)="predefineTextBlocks()" *ngIf="result">Add Text Blocks automatically.</a>
+                <a
+                    class="alert-link"
+                    jhiTranslate="artemisApp.textAssessment.predefineTextBlocks"
+                    (click)="predefineTextBlocks()"
+                    *ngIf="result && !exercise.automaticAssessmentEnabled"
+                    >Add Text Blocks automatically.</a
+                >
             </div>
         </div>
         <div *ngFor="let assessment of referencedFeedback; let i = index" class="col-12 col-md-6 col-xl-4">
             <jhi-text-assessment-detail
                 [(assessment)]="referencedFeedback[i]"
+                [block]="referencedTextBlocks[i]"
```

Comment:
Is it always guaranteed that the `referencedFeedback` and `referencedTextBlocks` have the same length?
Otherwise this would cause an index out of bounds.

Maybe a different type of data structure would be useful (dictionary or associative list).

Reply:
Yes, we guarantee equal length for `referencedFeedback` and `referencedTextBlocks`.

I agree, that we should introduce a better data structure for this. I suggest to improve this in a future iteration, which will include updating the text assessment user interface to be more similar with the text-result view.